src/util.cpp

```diff
@@ -889,6 +889,99 @@ QSize CMinimumStackedLayout::sizeHint() const
 * Other Classes                                                                *
 \******************************************************************************/
 // Network utility functions ---------------------------------------------------
+bool NetworkUtil::ParseNetworkAddressString ( QString strAddress, QHostAddress& InetAddr, bool bEnableIPv6 )
+{
+    // try to get host by name, assuming
+    // that the string contains a valid host name string or IP address
+    const QHostInfo HostInfo = QHostInfo::fromName ( strAddress );
+
+    if ( HostInfo.error() != QHostInfo::NoError )
+    {
+        // qInfo() << qUtf8Printable ( QString ( "Invalid hostname" ) );
+        return false; // invalid address
+    }
+
+    foreach ( const QHostAddress HostAddr, HostInfo.addresses() )
+    {
+        // qInfo() << qUtf8Printable ( QString ( "Resolved network address to %1 for proto %2" ) .arg ( HostAddr.toString() ) .arg (
+        // HostAddr.protocol() ) );
+        if ( HostAddr.protocol() == QAbstractSocket::IPv4Protocol || ( bEnableIPv6 && HostAddr.protocol() == QAbstractSocket::IPv6Protocol ) )
+        {
+            InetAddr = HostAddr;
+            return true;
+        }
+    }
+    return false;
+}
+
+#ifndef CLIENT_NO_SRV_CONNECT
+bool NetworkUtil::ParseNetworkAddressSrv ( QString strAddress, CHostAddress& HostAddress, bool bEnableIPv6 )
+{
+    // init requested host address with invalid address first
+    HostAddress = CHostAddress();
+
+    QRegularExpression plainHostRegex ( "^([^\\[:0-9.][^:]*)$" );
+    if ( plainHostRegex.match ( strAddress ).capturedStart() != 0 )
+    {
+        // not a plain hostname? then don't attempt SRV lookup and fail
+        // immediately.
+        return false;
+    }
+
+    QDnsLookup* dns = new QDnsLookup();
+    dns->setType ( QDnsLookup::SRV );
+    dns->setName ( QString ( "_jamulus._udp.%1" ).arg ( strAddress ) );
+    dns->lookup();
+    // QDnsLookup::lookup() works asynchronously. Therefore, wait for
+    // it to complete here by resuming the main loop here.
+    // This is not nice and blocks the UI, but is similar to what
+    // the regular resolve function does as well.
+    QTime dieTime = QTime::currentTime().addMSecs ( DNS_SRV_RESOLVE_TIMEOUT_MS );
+    while ( QTime::currentTime() < dieTime && !dns->isFinished() )
+    {
+        QCoreApplication::processEvents ( QEventLoop::ExcludeUserInputEvents, 100 );
+    }
+    QList<QDnsServiceRecord> records = dns->serviceRecords();
+    dns->deleteLater();
+    if ( records.length() != 1 )
+    {
+        return false;
+    }
+    QDnsServiceRecord record = records.first();
+    if ( record.target() == "." || record.target() == "" )
+    {
+        // RFC2782 says that "." indicates that the service is not available.
+        // (Qt strips the trailing dot, which is why we check for empty string
+        // as well)
```

Comment:
Does this mean we might have two dots? (When and why) is this valid? Why don't we check just for the empty string?

Reply:
The code basically covers both what the RFC states plus what I've practically observed during testing on Linux. As I haven't tested all other platforms or different DNS implementations, I'd go with this duplicate check -- it's about the failure path anyway. It might be sufficient to check for the empty string, but it would be some effort to prove, I guess?

Reply:
At least add a comment here that the dot check might not be needed.

Reply:
Done